docs/upgrade.txt

```diff
@@ -11,109 +11,39 @@ Upgrade Library Version
 .. meta::
    :keywords: php framework, odm, code example
 
-.. contents:: On this page
-   :local:
-   :backlinks: none
-   :depth: 1
-   :class: singlecol
+The PHP library uses `semantic versioning <https://semver.org/>`__. Upgrading
+to a new major version may require changes to your application.
 
-Overview
---------
+Upgrading from version 3 to 4
+-----------------------------
 
-On this page, you can learn how to upgrade {+odm-short+} to a new major version.
-This page also includes the changes you must make to your application to upgrade
-your object-document mapper (ODM) version without losing functionality, if applicable.
+- Laravel 10.x is required
 
-How to Upgrade
---------------
+- Change dependency name in your composer.json to ``"mongodb/laravel-mongodb": "^4.0"``
+  and run ``composer update``
 
-Before you upgrade, perform the following actions:
+- Change namespace from ``Jenssegers\Mongodb\`` to ``MongoDB\Laravel\``
+  in your models and config
 
-- Ensure the new library version is compatible with the MongoDB Server version
-  your application connects to and the version of Laravel that your
-  application runs on. See the :ref:`<laravel-compatibility>`
-  page for this information.
-- Address any breaking changes between the version of {+odm-short+} that
-  your application currently uses and your planned upgrade version in the
-  :ref:`<laravel-breaking-changes>` section of this guide.
+- Remove support for non-Laravel projects
 
-To upgrade your library version, run the following command in your application's 
-directory:
+- Replace ``$dates`` with ``$casts`` in your models
 
-.. code-block:: bash
-   
-    composer require mongodb/laravel-mongodb:{+package-version+}
+- Call ``$model->save()`` after ``$model->unset('field')`` to persist the change
 
-To upgrade to a different version of the library, replace the information after
-``laravel-mongodb:`` with your preferred version number.
+- Replace calls to ``Query\Builder::whereAll($column, $values)`` with
+  ``Query\Builder::where($column, 'all', $values)``
 
-.. _laravel-breaking-changes:
+- ``Query\Builder::delete()`` doesn't accept ``limit()`` other than ``1`` or ``null``.
 
-Breaking Changes
-----------------
+- ``whereDate``, ``whereDay``, ``whereMonth``, ``whereYear``, ``whereTime``
+  now use MongoDB operators on date fields
```

Comment:
The changes in this file seems unrelated to the PR.

Reply:
Thanks for catching this -- I'll copy in the file from the 4.1 branch since it's not relevant to my changes.